projects/components/src/stepper/stepper.component.ts

```diff
@@ -104,26 +107,35 @@ export class StepperComponent implements AfterContentInit {
 
   public steps$?: Observable<StepperTabComponent[]>;
 
+  public constructor(private readonly cdr: ChangeDetectorRef, private readonly subs: SubscriptionLifecycle) {}
+
   public ngAfterContentInit(): void {
     this.steps$ = queryListAndChanges$(this.steps).pipe(map(list => list.toArray()));
+    this.subs.add(
+      merge(
+        this.selectionChange.pipe(map(({ selectedStep }) => selectedStep)),
```

Comment:
This part looks fine (although more complicated than I was intending! I was just thinking each step component can watch itself so you don't need to do the change merging here).